crates/flux-refineck/src/checker.rs

```diff
@@ -509,9 +509,15 @@ impl<'ck, 'genv, 'tcx, M: Mode> Checker<'ck, 'genv, 'tcx, M> {
                 span,
             )?;
             bug::track_span(span, || {
-                dbg::statement_span!("start", stmt, infcx, env, span);
+                dbg::statement!("start", stmt, infcx, env, span);
                 self.check_statement(&mut infcx, &mut env, stmt)?;
-                dbg::statement_span!("end", stmt, infcx, env, span);
+                dbg::statement!(
+                    "end",
+                    stmt,
+                    RefineCtxtTrace::new(&infcx),
+                    TypeEnvTrace::new(&env),
```

Comment:
I'd put this inside the macro. The `Checker` doesn't care about how these are dumped.